controllers/status.go

```diff
@@ -262,3 +316,15 @@ func removeMCPDuplicateEntries(mcps []mcov1.MachineConfigPool) []mcov1.MachineCo
 
 	return filtered
 }
+
+func removeUnMatchedTunedProfiles(nodes []corev1.Node, profiles []tunedv1.Profile) []tunedv1.Profile {
+	filteredProfiles := make([]tunedv1.Profile, 0)
+	for _, profile := range profiles {
+		for _, node := range nodes {
+			if profile.Name == node.Name {
+				filteredProfiles = append(filteredProfiles, profile)
+			}
```

Comment:
nit: in general you can add break here, it not possible that you will have two nodes with the same name